app/facades/npc_facade.rb

```diff
@@ -4,111 +4,99 @@ def self.create_npc(score_type, level)
+                           dex_attack: base_info.stats.dex_attack,
+                           str_attack: base_info.stats.str_attack,
+                           skills_acrobatics: base_info.stats.skills.skills[:acrobatics],
+                           skills_animal_handling: base_info.stats.skills.skills[:animal_handling],
+                           skills_arcana: base_info.stats.skills.skills[:arcana],
+                           skills_athletics: base_info.stats.skills.skills[:athletics],
+                           skills_deception: base_info.stats.skills.skills[:deception],
+                           skills_history: base_info.stats.skills.skills[:history],
+                           skills_insight: base_info.stats.skills.skills[:insight],
+                           skills_intimidation: base_info.stats.skills.skills[:intimidation],
+                           skills_investigation: base_info.stats.skills.skills[:investigation],
+                           skills_medicine: base_info.stats.skills.skills[:medicine],
+                           skills_nature: base_info.stats.skills.skills[:nature],
+                           skills_perception: base_info.stats.skills.skills[:perception],
+                           skills_performance: base_info.stats.skills.skills[:performance],
+                           skills_persuasion: base_info.stats.skills.skills[:persuasion],
+                           skills_religion: base_info.stats.skills.skills[:religion],
+                           skills_sleight_of_hand: base_info.stats.skills.skills[:sleight_of_hand],
+                           skills_stealth: base_info.stats.skills.skills[:stealth],
+                           skills_survival: base_info.stats.skills.skills[:survival],
+                           spell_slots: base_info.spells.spell_slots.instance_of?(String) ? base_info.spells.spell_slots : base_info.spells.spell_slots.to_s,
+                           sub_ancestry: base_info.sub_ancestry.instance_of?(String) ? base_info.sub_ancestry : base_info.sub_ancestry[:name],
+                           traits: base_info.traits.to_s,
+                           vision: base_info.vision,
+                           is_saved: 0)
+    NpcModelPack.create(npc_model_id: npc.id, pack_id: pack.id)
+    armor.each do |piece|
+      NpcModelArmor.create(npc_model_id: npc.id, armor_id: piece.id) unless piece == ''
     end
-    @weapons.each do |weapon|
-      NpcModelWeapon.create(npc_model_id: npc[0].id, weapon_id: weapon.id)
+    weapons.each do |weapon|
+      NpcModelWeapon.create(npc_model_id: npc.id, weapon_id: weapon.id)
     end
-    if @spells.class != String
-      @spells.flatten.each do |spell|
-        NpcModelSpell.create(npc_model_id: npc[0].id, spell_id: spell.id) unless spell.nil?
+    if spells.class != String
+      spells.flatten.each do |spell|
+        NpcModelSpell.create(npc_model_id: npc.id, spell_id: spell.id) unless spell.nil?
       end
     end
     npc
   end
```

Comment:
All of this updated so all that is getting sent to the NpcModelController is the npc itself, since the spells, weapons, packs, and armor are all already available via model relations